app/models/Tokens.py

```diff
@@ -0,0 +1,191 @@
+    """
+    AUTH_TOKEN:    Final[TokenTypes] = TokenAttributes(TokenTypes.AUTH_TOKEN)
+    REFRESH_TOKEN: Final[TokenTypes] = TokenAttributes(TokenTypes.REFRESH_TOKEN)
+
+
+# -------- Classic models -------- #
+class Token:
+    value: str | None = None
+    attributes: TokenAttributes
+
+    def __init__(self, attributes: TokenAttributes, value: str | None = None):
+        self.attributes = attributes
+        self.value      = value
+
+    def generate(self, user_id: int):
+        """
+        This method generates a single token.
+        It uses the attributes specified to generate the correct token using the correct private key.
+        :return: A Json Web Token.
+        """
+        # Generating data for the token
+        creation_date = datetime.now()
+        expire_date = datetime.now() + timedelta(minutes=float(self.attributes.expire_time))
+        jwt_data = {
+            "user_id": user_id,
+            "salt": str(bcrypt.gensalt()),
+            "iat": creation_date,
+            "exp": expire_date
+        }
+        self.value = jwt.encode(jwt_data, str(self.attributes.secret), self.attributes.algorithm)
+
+    async def revoke(self, redis_db: redis.Redis = Redis.get_redis()):
+        """
+        This method handles the necessary operations to revoke the token.
+        """
+        if self.value is None:
+            return
+
+        # Extracting payload
+        data = await self.extract_payload()
+        # Adding the token to redis blocklist with an expiration date according to the expiration date of the token.
+        redis_db.setex(self.value, data["exp"], self.attributes.token_type)
+        self.value = None
+
+
+    async def extract_payload(self):
+        """
+        This method extracts the payload from the current token.
+        """
+
+        if self.value is None:
+            return None
+
+        # We check if the token has been revoked
+        if await self.is_revoked():
+            raise HTTPException(status_code=401, detail=CommonErrorMessages.TOKEN_REVOKED)
+
+        # We try to extract the payload from the token
+        try:
+            # We use a different key whether it is a Refresh or an Auth token.
+            # Both are supplied in the ENV variables
+            data = jwt.decode(jwt=self.value, key=str(self.attributes.secret), algorithms=[self.attributes.algorithm])
+
+        except jwt.exceptions.ExpiredSignatureError:
+            # This exception is raised if the date of the token has expired
+            raise HTTPException(status_code=401, detail=CommonErrorMessages.TOKEN_EXPIRED)
+
+        except jwt.exceptions.InvalidTokenError:
+            # This exception is raised if the token cannot be decoded.
+            raise HTTPException(status_code=401, detail=CommonErrorMessages.TOKEN_INVALID)
+
+        # We return the payload
+        return data
+
+
+    async def is_revoked(self, redis_db: redis.Redis = Redis.get_redis()):
```

Comment:
should be typed with Awaitable[<ReturnType>] or Coroutine[<YieldType>, <SendType>, <ReturnType>]